packages/python-packages/protocol-stub-generator/protocol/parse_yml.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import yaml
+import json
+import argparse
+import logging
+import os
+import tempfile
+from .protocol_models import ProtocolClientView
+
+class LLCGenerator:
+    def __init__(self):
+        parser = argparse.ArgumentParser(
+            description="Parse a python package and generate json token file to be supplied to API review tool"
+        )
+        parser.add_argument(
+            "--pkg-path",
+            required=True,
+            help=("Package root path"),
+        )
+        parser.add_argument(
+            "--temp-path",
+            help=("Temp path to extract package"),
+            default=tempfile.gettempdir(),
+        )
+        parser.add_argument(
+            "--out-path",
+            default=os.getcwd(),
+            help=("Path to generate json file with parsed tokens"),
+        )
+        parser.add_argument(
+            "--verbose",
+            help=("Enable verbose logging"),
+            default=False,
+            action="store_true",
+        )
+
+        parser.add_argument(
+            "--hide-report",
+            help=("Hide diagnostic report"),
+            default=False,
+            action="store_true",
+        )
+
+        parser.add_argument(
+            "--filter-namespace",
+            help=("Generate Api view only for a specific namespace"),
+        )
+
+        args = parser.parse_args()
+        if not os.path.exists(args.pkg_path):
+            logging.error("Package path [{}] is invalid".format(args.pkg_path))
+            exit(1)
+        elif not os.path.exists(args.temp_path):
+            logging.error("Temp path [{0}] is invalid".format(args.temp_path))
+            exit(1)
+
+        self.pkg_path = args.pkg_path
+        self.temp_path = args.temp_path
+        self.out_path = args.out_path
+        self.hide_report = args.hide_report
+        if args.verbose:
+            logging.getLogger().setLevel(logging.DEBUG)
+
+        self.filter_namespace = ""
+        if args.filter_namespace:
+            self.filter_namespace = args.filter_namespace
+
+    def parse_yaml(self):
+        # open the yaml file and load data
+        with open(self.pkg_path) as f:
+            data = yaml.safe_load(f)
+
+        create_python_name(data)
+
+        # Create main_view for LLC API View
+        main_view = ProtocolClientView.from_yaml(data)
+
+        # Write output to .json
+        out_file = self.pkg_path.split("\\")
+        file_name = out_file[len(out_file)-1]
+        j = open(self.out_path + "\\"+ file_name.split(".")[0]+".json", "w")
+        parsed_json = json.dumps(main_view.to_json(), default=lambda o: o.__dict__)
+        j.write(parsed_json)
```

Comment:
@kristapratico is probabaly a better person to answer here. But I have seen python code using "with open" when dealing with file read write and code will look a bit more cleaner since you don't have to write file.close() explicitly.
Something like as follows:
```
with open(out_file_path, "w") as json_file:
        json_file.write(json_tokens)

Reply:
agreed